include/kernels/multi_reduce_core.cuh

```diff
@@ -13,6 +13,19 @@ namespace quda
   namespace blas
   {
 
+    /**
+       @brief Return the batch block size used for multi reductions.
+       For now, we leave this at 1 unless fast-compilation is enabled,
+       since it provides negligible benefit (but substantially longer
+       tuning).  When fast-compilation is enabled, the increased batch
+       count can dramatically improve performance.
+     */
+#ifndef QUDA_FAST_COMPILE_REDUCE
```

Comment:
should this be #ifdef (set to 1 if fast compile, 8 otherwise)?

Reply:
No.  😄   As the comment notes, the perf benefit from block batching is non-existent when fast-compile is disabled.  However, when enabled, block batching allows us to increase the occupancy, and helps to close the gap relative to fast-compile off.

Reply:
One could argue I suppose that we should just leave it on always then, but enabling it for fast compile off has an adverse impact on auto tuning time.  Hence the above code.

Long-term plan (post PR merge) is to rewrite this routine to utilize 2-d hierarchical tiling as opposed to the approach taken presently (which uses register tiling in the x dimension, and cache tiling in the y dimension).  Once I have done that, the block batch tuning will be have a significant benefit then.